drivers/ds323x/ds3234.c

```diff
@@ -94,6 +95,12 @@ int ds3234_pps_init(const ds3234_params_t *dev)
     reg &= ~(DS323X_REG_CONTROL_EOSC_MASK | DS323X_REG_CONTROL_INTCN_MASK |
         DS323X_REG_CONTROL_RS1_MASK | DS323X_REG_CONTROL_RS2_MASK);
     ds3234_write_reg(dev, DS323X_REG_CONTROL, 1, &reg);
+    uint8_t readback = 0;
```

Comment:
I tested this and still does not solve the issue of not having the sensor connected. During the tests I disconnected the sensor and sometimes the reading of the control register returned 0, so the check would pass. I suggest to write the control register first with two known bits and then checking. Something like:
```C
    ds3234_read_reg(dev, DS323X_REG_CONTROL, 1, &reg);

    reg |= DS323X_REG_CONTROL_RS1_MASK;
    reg &= ~(DS323X_REG_CONTROL_RS2_MASK);

    ds3234_write_reg(dev, DS323X_REG_CONTROL, 1, &reg);

    uint8_t readback = 0;
    ds3234_read_reg(dev, DS323X_REG_CONTROL, 1, &readback);
    if (reg != readback) {
        DEBUG("ds3234: readback mismatch: expected %u, actual %u\n", (unsigned)reg, (unsigned)readback);
        return -EIO;
    }

    reg &= ~(DS323X_REG_CONTROL_EOSC_MASK | DS323X_REG_CONTROL_INTCN_MASK |
        DS323X_REG_CONTROL_RS1_MASK | DS323X_REG_CONTROL_RS2_MASK);
    ds3234_write_reg(dev, DS323X_REG_CONTROL, 1, &reg);
```